src/sage/combinat/permutation.py

```diff
@@ -378,6 +393,29 @@ class Permutation(CombinatorialElement):
         sage: type(p)
         <class 'sage.combinat.permutation.StandardPermutations_n_with_category.element_class'>
 
+    Construction using the Steinhaus-Johnson Trotter algorithm to generate
+    the permutations (lexicographic algorithm by default)::
+
+        sage: p = Permutation([1, 2, 3, 4], algorithm='sjt'); p
+        [1, 2, 3, 4]
+        sage: p = p.next(); p
+        [1, 2, 4, 3]
+        sage: p = p.next(); p
+        [1, 4, 2, 3]
+        sage: p = Permutation([1, 2, 3], algorithm='sjt')
+        sage: for _ in range(6):
+        ....:     p = p.next()
+        ....:
+        sage: p
+        False
```

Comment:
Replace this with `[next(p) for  _ in range(6)]` directly?

Reply:
Permutation objects need reassignment to go to the next permutation, this wouldn't work and loop on the same one. I think this is rather fine and self-explanatory.

Reply:
`[(p := next(p)) for  _ in range(6)]` works (replace line 404 with `p = ...; p`)

Reply:
I'm not a fan of this anyway, the loop looks more verbose which is what we want for examples.